doc/man/bee.1
.\" Automatically generated by Pod::Man 2.27 (Pod::Simple 3.28)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{
.    if \nF \{
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "bee 1"
.TH bee 1 "2015-05-24" "" "Lua application server and NoSQL database"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
bee\- a Lua application server and a NoSQL database
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
.Vb 1
\&    bee [OPTIONS]
.Ve
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
Bee is a Lua application server and a NoSQL database.
.PP
Its key properties include:
.IP "\(bu" 4
Licence: simplified \s-1BSD\s0
.IP "\(bu" 4
All data is maintained in \s-1RAM\s0
.IP "\(bu" 4
Data persistence is implemented using Write Ahead Log and snapshotting
.IP "\(bu" 4
Supports asynchronous replication and hot standby
.IP "\(bu" 4
Uses coroutines and asynchronous I/O to implement
high-performance lock-free access to data
.IP "\(bu" 4
Available on Linux, FreeBSD, Mac \s-1OS X\s0
.IP "\(bu" 4
Stored procedures in Lua are supported.
.SS "Data model"
.IX Subsection "Data model"
The basic atom of storage in Bee is called tuple. A tuple
can have arbitrary number of fields. The first field or fields 
in the tuple
are always the identifying unique key. Tuples form spaces. It is
possible to search tuples in spaces using the primary or secondary
keys. It is possible to change, as well as add or remove fields.
.SH "Command-line options"
.IX Header "Command-line options"
.IP "\-h, \-\-help" 4
.IX Item "-h, --help"
Displays helpscreen and exits.
.IP "\-V, \-\-version" 4
.IX Item "-V, --version"
Prints program version and exits.


debian/changelog
bee (1.6.5.20.g51b2e0d-1) unstable; urgency=medium

  * Remove bee-mysql-module and bee-pg-module.

 -- Roman Tsisyk <roman@bee.org>  Tue, 28 Apr 2015 14:34:17 +0300

bee (1.6.3.539.g211672f-1) unstable; urgency=medium

  * Init infrastructure for bee 1.6.
  * Update debian/copyright:
     + cmake/luatest.cpp
     + third_party/PMurHash.*
     + third_party/lua-yaml/*
     + third_party/sha1.*
     + third_party/open_memstream.*
  * Add lintian-oferrides: libyaml:
    bee has patched version of the library.
  * Rename -plugin* package to -module*
  * Conflicts: bee 1.5 (it was renamed to bee-lts).
  * NOSTRIP /usr/bin/bee binary: it uses symbols runtime.

 -- Dmitry E. Oboukhov <unera@debian.org>  Mon, 27 Oct 2014 14:33:07 +0300

bee (1.5.3.53.ge2308d8-1) unstable; urgency=medium

  * New Debian snapshot
   + add sham storage
   + some fixes in db.net.db
   + uses O_DIRECT options to create snapshots
   + update documentation/manpagses
   + memleak in bee client

 -- Dmitry E. Oboukhov <unera@debian.org>  Wed, 19 Feb 2014 18:37:26 +0400

bee (1.5.2.20.g5f5d924-1) unstable; urgency=low

  * Rebuild for Debian/unstable (beetar binary).

 -- Dmitry E. Oboukhov <unera@debian.org>  Sat, 21 Dec 2013 13:18:28 +0400

bee (1.5.2.16.g18b8e4e-1) unstable; urgency=low

  * New Debian snapshot.
   - Drop legacy packages (libbee*).
   - Drop bee-dbg.
   - Don't strip bee (it uses debug symbols runtime).

 -- Dmitry E. Oboukhov <unera@debian.org>  Fri, 20 Dec 2013 15:44:11 +0400

bee (1.5.1.302.gd906190-1) unstable; urgency=low

  * New Debian snapshot.
   - db.session.storage,
   - bugfixes

 -- Dmitry E. Oboukhov <unera@debian.org>  Mon, 09 Dec 2013 16:23:49 +0400

bee (1.5.1.218.g1a69fd6-1) unstable; urgency=low

  * New Debian snapshot
   - patch for OpenVZ (disable MAP_SHARED mmap flag with OpenVZ).
   - fix socket:readline(..., timeout) bug.
   - suggests -dbg version of package.

 -- Dmitry E. Oboukhov <unera@debian.org>  Tue, 19 Nov 2013 16:17:23 +0400

bee (1.5.1.150.g029f82f-1) unstable; urgency=low

  * New Debian snapshot (fix memory corruption).

 -- Dmitry E. Oboukhov <unera@debian.org>  Fri, 25 Oct 2013 15:26:36 +0400

bee (1.5.1.148.g6dddbc7-1) unstable; urgency=low

  * New Debian snapshot.

 -- Dmitry E. Oboukhov <unera@debian.org>  Wed, 23 Oct 2013 13:46:46 +0400

bee (1.5.1.84.g9f2e646-1) unstable; urgency=low

  * Drop BSD dependencies in Debian/kFreeBSD (try to fix FTBFS with Debian/BSD).

 -- Dmitry E. Oboukhov <unera@debian.org>  Mon, 09 Sep 2013 15:52:39 +0400

bee (1.5.1.82.ga064ebe-1) unstable; urgency=low

  * Remove bee/config.h pointedly (try to fix FTBFS with Debian/kFreeBSD).

 -- Dmitry E. Oboukhov <unera@debian.org>  Mon, 09 Sep 2013 13:35:06 +0400

bee (1.5.1.81.g4382f16-1) unstable; urgency=low

  * Fix debian/rules clean section (cleanup CMake files).

 -- Dmitry E. Oboukhov <unera@debian.org>  Sun, 08 Sep 2013 23:05:29 +0400

bee (1.5.1.80.gc84e35c-1) unstable; urgency=low

  * New snapshot (try to fix FTBFS with Debian/kFreeBSD).

 -- Dmitry E. Oboukhov <unera@debian.org>  Sun, 08 Sep 2013 22:14:09 +0400

bee (1.5.1.78.g2e82d26-1) unstable; urgency=low

  * New snapshot. Integration upload.
   - Add missing copyright holders, closes: #722059
   - Drop non-free materials from tarbal, closes: #722060
   - Fix memory leak in db.ipc.channel:put.

 -- Dmitry E. Oboukhov <unera@debian.org>  Sun, 08 Sep 2013 02:00:51 +0400

bee (1.5.1.76.g5ed3fbe-1) unstable; urgency=low

  * New snapshot.
   - Fix memory leak in db.ipc.channel:put(v, 0)

 -- Dmitry E. Oboukhov <unera@debian.org>  Thu, 05 Sep 2013 14:36:52 +0400

bee (1.5.1.73.gd1b3db6-1) unstable; urgency=low

  * New snapshot
   - Fix crashes with broken snapshot.
   - Cleanup debian/copyright (drop php-connector section).
   - Drop empty TODO files from binary packages.
   - debian/* relicensed as bee.
   - bee-plugins has 'any' architecture: it depends on plugins
     that contain ELF-files.
   - Fix FTBFS with Debian/kFreeBSD, closes: #715358.
   - Drop irrelevant debian/README.source.
   - Add postgresql-plugin package.
   - Add mysql-plugin package.
   - Add -plugins package (Now it depends on other plugin. Then it will
     contain some internal bee plugins).
   - Remove non-free rfc4627.txt.
   - Update debian/copyright.

 -- Dmitry E. Oboukhov <unera@debian.org>  Wed, 04 Sep 2013 12:45:34 +0400

bee (1.4.9+20130611.2012-1) unstable; urgency=low

  * New snapshot, fix crash with gcc-4.7.3.

 -- Dmitry E. Oboukhov <unera@debian.org>  Tue, 11 Jun 2013 20:12:22 +0400

bee (1.4.9+20130608.2059-1) unstable; urgency=low

  * New upstream release: move to C++ from ObjectC. Add db.cjson module.

 -- Dmitry E. Oboukhov <unera@debian.org>  Sat, 08 Jun 2013 20:59:30 +0400

bee (1.4.9+20130415.1808-1) unstable; urgency=low

  * New upstream release (fix disconnect bug).
    https://bugs.launchpad.net/bee/+bug/1168076

 -- Dmitry E. Oboukhov <unera@debian.org>  Mon, 15 Apr 2013 18:08:14 +0400

bee (1.4.9-1) unstable; urgency=low

  * New upstream release.
   + Apply patch from Colin Watson <cjwatson@ubuntu.com>, closes: #703044,
   + Fix VCS-link, closes: #681703
   + Fix FTBFS (kfreebsd), patch from Steven Chamberlain <steven@pyro.eu.org>,
     closes: 695541

 -- Dmitry E. Oboukhov <unera@debian.org>  Wed, 10 Apr 2013 12:08:49 +0400

bee (1.4.8+20130306.1415-1) unstable; urgency=low

  * New snapshot (some fixes in iproto).

 -- Dmitry E. Oboukhov <unera@debian.org>  Wed, 06 Mar 2013 14:15:10 +0400

bee (1.4.8+20130212.1728-1) unstable; urgency=low

  * Rebuilt last snapshot for Debian.

 -- Dmitry E. Oboukhov <unera@debian.org>  Tue, 12 Feb 2013 17:27:48 +0400

bee (1.4.8+20130212.1647-1) nowtaxi; urgency=low

  * New snapshot (fix async I/O bugs).

 -- Roman V. Nikolaev <rshadow@rambler.ru>  Tue, 12 Feb 2013 16:47:10 +0400

bee (1.4.8+20130122.1705-1) unstable; urgency=low

  * New snapshot (fix spam in bee.log).

 -- Dmitry E. Oboukhov <unera@debian.org>  Tue, 22 Jan 2013 17:05:18 +0400


extra/bee.rb
require 'formula'

class Bee < Formula
  homepage 'http://bee.org'

  depends_on 'cmake'    => :build
  depends_on "readline" => :build

  option 'with-debug', "Build Debug version"
  option 'with-tests', "Run Tests after building"

  stable do
    url 'https://github.com/bee/bee.git', :branch => "stable", :shallow => false
    depends_on 'e2fsprogs' => :recommended
    if build.with? 'tests'
      depends_on 'python-daemon' => [:python, "daemon",  :build]
      depends_on 'pyyaml'        => [:python, "yaml",    :build]
      depends_on 'pexpect'       => [:python, "pexpect", :build]
    end
    version "1.5"
  end

  devel do
    url 'https://github.com/bee/bee.git', :branch => "master", :shallow => false
    depends_on 'e2fsprogs' => :build
    if build.with? 'tests'
      depends_on 'python-daemon' => [:python, "daemon", :build]
      depends_on 'pyyaml'        => [:python, "yaml",   :build]
    end
    version "1.6"
  end

  def install
    args = []
    if build.with? 'debug'
      ENV.enable_warnings
      ENV.deparallelize
      args << "-DCMAKE_BUILD_TYPE=Debug"
      ohai "Building with Debug"
    else
      args << "-DCMAKE_BUILD_TYPE=Release"
      ohai "Building with Release"
    end
    args << "-DENABLE_CLIENT=True" if build.stable?
    args << "-DCMAKE_INSTALL_SYSCONFDIR=#{prefix}/etc"
    args << "-DCMAKE_INSTALL_LOCALSTATEDIR=#{prefix}/var"
    args << "-DREADLINE_ROOT=/usr/local/Cellar"
    args += std_cmake_args

    ohai "Preparing"
    version = `git -C #{cached_download} describe HEAD`

    File.open(buildpath/"VERSION", 'w') {|file| file.write(version)}

    ohai "Configuring:"
    system "cmake", ".", *args

    ohai "Building:"
    system "make"

    ohai "Installing:"
    system "make install"

    ohai "Installing man"
    man1.install 'doc/man/bee.1'
    if build.stable?
      man1.install 'doc/man/bee_db.1'
    end

    ohai "Installing config"
    if build.stable?
      inreplace prefix/"etc/bee.cfg", /^work_dir =.*/, "work_dir = #{prefix}/var/lib/bee"
    else
      doc.install "test/db/db.lua"
      inreplace doc/"db.lua" do |s|
          s.gsub!(/^os = require.*\n/    , '')
          s.gsub!(/os.getenv\("LISTEN"\)/, '3301')
          s.gsub!(/os.getenv\('ADMIN'\)/ , '3313')
          s.gsub!(/(rows_per_wal\s*=).*/ , '\1 500,')
          s.gsub!(/^}.*/                 , "\twork_dir\t\t\t= \"#{prefix}/var/lib/bee\",\n}")
      end
    end

    if build.with? 'tests'
        ohai "Testing Bee with internal test suite:"
        system "/usr/bin/env", "python", buildpath/"test/test-run.py", "--builddir", buildpath, "--vardir", buildpath/"test/var"
    end
  end

  test do
      system bin/"bee", "--version"
  end
end
